File: Week_01/plusOne/plusOne.go
func plusOne(digits []int) []int {
    add := 1
    for length := len(digits)-1; length >= 0 && add > 0; length-- {
        sum := digits[length] + add
        add = sum / 10
        digits[length] = sum % 10
    }
    
    if add > 0 {
        digits = append([]int{add}, digits...)
    }
    return digits
}

File: Week_01/plusOne/plusOne.java
class Solution {
    public int[] plusOne(int[] digits) {
        for(int cur = digits.length - 1; cur >= 0; cur--) {
            digits[cur]++;
            if(digits[cur] != 10) {
                return digits;
            } else {
                digits[cur] = 0;
            }
        }
        digits = new int[digits.length + 1];
        digits[0] = 1;
        return digits;
    }
}